operations/iuf/stages/management_nodes_rollout.md

```diff
@@ -26,22 +35,184 @@ The following arguments are most often used with the `management-nodes-rollout`
+[update-cfs-config](../stages/update_cfs_config.md) stage.
+
+## Manually upgrade or rebuild NCN worker node with specific image and CFS configuration outside of IUF
+
+**NOTE** This section describes how to manually rebuild/upgrade a worker node outside of IUF with an image and CFS configuration created through IUF. **This is not the normal procedure that IUF uses for rebuilding/upgrading NCN
+worker nodes.** This procedure should be followed if NCN worker nodes need to be rebuilt or upgraded outside of IUF.
+
+The upgrade and rebuild procedures for NCN worker nodes are identical. These instructions apply to both NCN worker node upgrades and NCN worker node rebuilds.
+The words 'rebuild' and 'upgrade' are exchangeable in this section.
+
+1. Get the image ID and CFS configuration created for worker nodes during the `prepare-images` and `update-cfs-config` stages. Follow the instructions in the [`prepare-images` Artifacts created](prepare_images.md#artifacts-created)
+documentation to get the values for `final_image_id` and `configuration` for images with a `configuration_group_name` value matching `Management_Worker`. These values will be used in the next step.
+
+1. Upgrade/rebuild the worker node. The worker node is automatically rebuilt using Argo workflows. If rebuilding multiple worker nodes at once, see [this page](../../node_management/Rebuild_NCNs/Rebuild_NCNs.md#restrictions) for restrictions.
+
+    (`ncn-m001#`) Rebuild a worker node. Use the values acquired in the previous step in place of `<final_image_id>` and `<configuration>`.
+
+    ```bash
+    /usr/share/doc/csm/upgrade/scripts/upgrade/ncn-upgrade-worker-storage-nodes.sh ncn-w001 --image-id <final_image_id> --desired-cfs-conf <configuration>
+    ```
+
+## Action needed if a worker rebuild fails
+
+In general, worker node rebuilds should complete successfully before starting another rebuild.
+A worker node can get into a bad state if it has been partially rebuilt and an attempt is made to restart the rebuild on that same node.
+As a result, it is not possible to start another worker node rebuild if there is an existing incomplete worker node rebuild workflow, where "incomplete" means it has stopped before successfully completing the full workflow.
+If an incomplete workflow exists and an attempt is made to start another worker rebuild workflow,
+the original incomplete worker rebuild workflow will continue and no new workflow will be created.
 
-When NCN management nodes are being rebuilt, an additional Argo workflow will execute in addition to the standard IUF Argo workflows. It will include the string `ncn-lifecycle-rebuild` in its name.
+If it is necessary to start an entirely new worker rebuild workflow after a previous worker rebuild workflow failed, the failed workflow must be deleted from Kubernetes first.
```

Comment:
I think we mean the failed workflow needs to be deleted from Argo, correct?

Reply:
Per the example in the doc showing how to do the `kubectl delete` and Atif's comments about why that is, I will leave this as is.